django/core/management/__init__.py

```diff
@@ -378,6 +385,24 @@ def execute(self):
         except CommandError:
             pass  # Ignore any option errors at this point.
 
+        # Read the settings module from the local pyproject.toml if present.
+        pyproject = Path("pyproject.toml")
+        if pyproject.is_file():
+            with pyproject.open("rb") as f:
+                _config = tomllib.load(f)
+            _settings_module = (
+                _config.get("tool", {}).get("django", {}).get("settings_module")
+            )
+            if _settings_module is not None:
+                os.environ.setdefault("DJANGO_SETTINGS_MODULE", _settings_module)
+                if _settings_module == os.environ["DJANGO_SETTINGS_MODULE"]:
+                    # The runserver command starts up a subprocess that has
+                    # DJANGO_SETTINGS_MODULE set, so we can't only add to
+                    # the path if DJANGO_SETTINGS_MODULE is not already set.
```

Comment:
This likely can be addressed differently. I intend to take a look at the `runserver` command to see how it spawns its subprocesses to see if we can handle this more simply here.